internal/component/loki/source/syslog/types.go

```diff
@@ -52,6 +68,15 @@ func (sc ListenerConfig) Convert() *scrapeconfig.SyslogTargetConfig {
 		lbls[model.LabelName(k)] = model.LabelValue(v)
 	}
 
+	var syslogFormat scrapeconfig.SyslogFormat
+	if sc.SyslogFormat == SyslogFormatRFC3164 {
+		syslogFormat = scrapeconfig.SyslogFormatRFC3164
+	} else if sc.SyslogFormat == SyslogFormatRFC5424 {
+		syslogFormat = scrapeconfig.SyslogFormatRFC5424
+	} else {
+		panic("unknown syslog format " + sc.SyslogFormat)
```

Comment:
This can only happen if validation fails but this is called anyway...

Reply:
I think it's ok not to do a check for the values, and instead to assume that `sc` is an already well-formatted Alloy config. In the struct below, you could just do:
```go
scrapeconfig.SyslogFormat(sc.SyslogFormat)
```

Reply:
Then we're basically relying on the promtail strings to match and might as well use them directly?